read_buffer/src/table.rs

```diff
@@ -496,22 +459,40 @@ impl MetaData {
         }
     }
 
-    // Extract schema information for a set of columns.
-    fn schema_for_column_names(&self, names: &[ColumnName<'_>]) -> Vec<(&str, LogicalDataType)> {
+    // Extract schema information for a set of columns. If a column name does
+    // not exist within the `Table` schema it is ignored and not present within
+    // the resulting schema information.
+    fn schema_for_column_names(&self, names: &[ColumnName<'_>]) -> Vec<(String, LogicalDataType)> {
         names
             .iter()
-            .map(|&name| {
-                let (k, v) = self.column_types.get_key_value(name).unwrap();
-                (k.as_str(), *v)
+            .filter_map(|&name| match self.column_types.get_key_value(name) {
+                Some((name, data_type)) => Some((name.clone(), *data_type)),
+                None => None,
```

Comment:
You could probably get your functional programming weenie hat on and avoid the `None => None` with something like

```suggestion
            .filter_map(|&name| {
              self.column_types.get_key_value(name) {
                .map(|(name, data_type)| (name.clone(), *data_type))
              })
```

Reply:
The same thing applies below too

Reply:
I knew `None => None` was avoidable but at that time of night I couldn't remember how. I shall investigate, thanks. I think the above might be a bit off with the `.map` call still in a block.